src/main/java/redis/clients/jedis/MultiNodePipelineBase.java

```diff
@@ -104,22 +113,27 @@ public final void sync() {
         } catch (JedisConnectionException jce) {
           log.error("Error with connection to " + nodeKey, jce);
           // cleanup the connection
+          // TODO these operations not thread-safe
           pipelinedResponsesIterator.remove();
           connections.remove(nodeKey);
-          IOUtils.closeQuietly(connection);
         } finally {
-          countDownLatch.countDown();
+          IOUtils.closeQuietly(connection);
```

Comment:
Looking at the current connection handling model,  only broken connections are closed.
```suggestion
          IOUtils.closeQuietly(connection);
```
The current connection handling model suggests closing the connection when the pipeline itself is closed, and it make sense to me. We can have multiple `sync()` reusing the same connection. I think it should be preserved.